File: 402/solution.cpp
class Solution {
public:
    string removeKdigits(string num, int k) {
        string ans = "";
        
        for (char digit: num) {
            while (!ans.empty() && ans.back() > digit && k) {
                ans.pop_back();
                k--;
            }
            if (!ans.empty() || digit != '0') ans.push_back(digit);
        }
        
        while (k && !ans.empty()) {
            ans.pop_back();
            k--;
        }
        
        return ans.length() > 0 ? ans : "0"; 
    }
};

File: 402/solution.go
func removeKdigits(num string, k int) string {
	ans := ""
	
	for _, digit := range(num) {
			for k > 0 && len(ans) > 0 && int(ans[len(ans) - 1]) > int(digit) {
					ans = ans[:len(ans) - 1]
					k--
			}
			
			if digit != '0' || len(ans) != 0 {
					ans += string(digit)
			}
	}
	
	for k > 0 && len(ans) > 0 {
			ans = ans[:len(ans) - 1]
			k--
	}
	
	if len(ans) > 0 {
			return ans
	} else {
			return "0"
	}
}